src/usb_moded-dbus.c

```diff
@@ -159,7 +184,10 @@ static DBusHandlerResult msg_handler(DBusConnection *const connection, DBusMessa
 		else
 		{
 			/* error checking is done when setting configuration */
-			if(!set_mode_setting(config))
+			int ret = set_mode_setting(config);
+			if (ret == SET_CONFIG_UPDATED)
+				usb_moded_send_config_signal(MODE_SETTING_ENTRY, MODE_SETTING_KEY, config);
```

Comment:
Could the call to `usb_moded_set_config_signal()` be part of the code inside `set_mode_setting` or even `set_config_setting`? This way, we move the responsibility from the caller to the callee, and can do it in a central place (we might have to filter out situations where we don't broadcast changes, e.g. at the start(?), though).